src/org/labkey/test/pages/pipeline/PipelineStatusDetailsPage.java

```diff
@@ -0,0 +1,433 @@
+package org.labkey.test.pages.pipeline;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.Nullable;
+import org.labkey.api.collections.CaseInsensitiveHashSet;
+import org.labkey.api.pipeline.PipelineJob;
```

Comment:
Just an FYI, avoid using server APIs. On TeamCity, we pull the API jar from Artifactory to avoid building it before every test suite (some tests need access to the XML schema definitions). The version used by tests lags behind the actual server version to avoid breaking tests every time we bump versions for monthly releases so using server APIs too extensively could lead to test noise on TeamCity.
That being said, these are probably fine because they're unlikely to change in any way the tests would care about.